<launch>
	<!-- RTAB-MAP SENSORY-MOTOR VERSION -->
	<!-- WARNING : Database is automatically deleted on each startup -->
	<!--           See "delete_db_on_start" option below... -->
	
	<!-- Nodes -->	
	<!-- cmd_vel_a has priority on cmd_vel_b -->
	<node name="abtr_velocity" pkg="rtabmap" type="abtr_velocity">
	    <param name="commands_hz" value="1.0" type="double"/>
	    <param name="cmd_vel_a_buffered" value="true" type="bool"/>
		<remap from="cmd_vel_a" to="user/cmd_vel" />
	<!-- 	<remap from="cmd_vel_b" to="rtabmap/cmd_vel" /> -->
		<remap from="cmd_vel" to="cmd_vel" />
	</node>
	
	<node name="rtabmap_out" pkg="rtabmap" type="rtabmap_out">
	    <param name="commands_hz" value="1.0" type="double"/>
	    <param name="idle_null_commands_sent" value="false" type="bool"/>
		<remap from="rtabmap/cmd_vel" to="rtabmap/cmd_vel" />
		<remap from="rtabmap/info" to="rtabmap/info" />
		<remap from="rtabmap/info_x" to="rtabmap/info_x" />
	</node>
	
	<!-- args: "delete_db_on_start" and "udebug" -->
	<node name="rtabmap" pkg="rtabmap" type="rtabmap" output="screen" args="--delete_db_on_start">
		<remap from="sm_state" to="sm_state" />
		<remap from="rtabmap/info" to="rtabmap/info" />
		<remap from="rtabmap/info_x" to="rtabmap/info_x" />
	</node>

	<node name="rtabmap_in" pkg="rtabmap" type="rtabmap_in">
		<param name="nb_commands" value="1" type="int"/>
		<remap from="cmd_vel" to="cmd_vel" />
		<remap from="sm_state" to="sm_state" />
		<remap from="sensor_data" to="sensor_data" />
	</node>
	
	<node name="twist_to_sms" pkg="rtabmap" type="twist_to_sms">
		<remap from="cmd_vel" to="cmd_vel" />
		<remap from="sm_state" to="sensor_data" />
	</node>
	
	<node name="twist_to_turtle_vel" pkg="rtabmap" type="twist_to_turtle_vel">
	</node>
	<node name="turtlesim" pkg="turtlesim" type="turtlesim_node">
	</node>

</launch>